src/build_rock/configure/generate_build_matrix.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+import yaml
+import os
+import argparse
+import json
+from enum import Enum
+from typing import Optional
+from ...shared import github_output
+
+
+class MATRIX_NAMES(Enum):
+    RUNNER = "runner-build-matrix"
+    LPCI = "lpci-build-matrix"
+
+
+class MissingArchSupport(Exception):
+    pass
+
+
+def get_target_archs(rockcraft: dict) -> list:
+    """get list of target architectures from rockcraft project definition"""
+
+    rock_platforms = rockcraft["platforms"]
+
+    target_archs = set()
+
+    for platf, values in rock_platforms.items():
+
+        if isinstance(values, dict) and "build-for" in values:
```

Comment:
I believe "build-for" is correct here.

Reply:
Thanks :smiley: Do you know of any projects that currently support this cross compilation feature?

Reply:
Didn't see any yet.